seed/analysis_pipelines/pipeline.py

```diff
@@ -188,6 +297,15 @@ def stop(self):
             locked_analysis.end_time = tz.now()
             locked_analysis.save()
 
+    def delete(self):
```

Comment:
Any reason for the AnalysisPipeline to perform the delete (vs the Analysis object directly)?

Reply:
My general idea for the pipeline is to abstract all modifications to the analysis because things can get messy real quick. By forcing all external usages of the analysis to use the AnalysisPipeline we have more control over the expected behavior.